Type C4244 Build warnings on MSVC compiler

When building the code from master (at the time of writing commit 0343007), build warnings are encountered regarding truncating values from uint16_t to uint8_t:

For example:
```css
...\Source\Lib\Common\Codec\EbCodingLoop.c(1639): warning C4244: 'initializing': conversion from 'uint16_t' to 'uint8_t', possible loss of data 
...\Source\Lib\Common\Codec\EbCodingLoop.c(1905): warning C4244: 'initializing': conversion from 'uint16_t' to 'uint8_t', possible loss of data 
...\Source\Lib\Common\Codec\EbCodingLoop.c(2729): warning C4244: 'initializing': conversion from 'uint16_t' to 'uint8_t', possible loss of data 
```

--- a/Source/Lib/Common/Codec/EbCodingLoop.c
+++ b/Source/Lib/Common/Codec/EbCodingLoop.c
@@ -508,13 +508,13 @@ static void Av1EncodeLoop(
             residual16bit->stride_y,
             context_ptr->blk_geom->tx_width[cu_ptr->tx_depth][context_ptr->txb_itr],
             context_ptr->blk_geom->tx_height[cu_ptr->tx_depth][context_ptr->txb_itr]);
-        uint8_t  tx_search_skip_fag = (picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_ENC_DEC && (picture_control_set_ptr->parent_pcs_ptr->atb_mode == 0 || cu_ptr->prediction_mode_flag == INTER_MODE)) ? get_skip_tx_search_flag(
+        uint8_t  tx_search_skip_flag = (picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_ENC_DEC && (picture_control_set_ptr->parent_pcs_ptr->atb_mode == 0 || cu_ptr->prediction_mode_flag == INTER_MODE)) ? get_skip_tx_search_flag(
             context_ptr->blk_geom->sq_size,
             MAX_MODE_COST,
             0,
             1) : 1;
 
-        if (!tx_search_skip_fag) {
+        if (!tx_search_skip_flag) {
                 encode_pass_tx_search(
                     picture_control_set_ptr,
                     context_ptr,
@@ -929,13 +929,13 @@ static void Av1EncodeLoop16bit(
                 residual16bit->stride_y,
                 context_ptr->blk_geom->tx_width[cu_ptr->tx_depth][context_ptr->txb_itr],
                 context_ptr->blk_geom->tx_height[cu_ptr->tx_depth][context_ptr->txb_itr]);
-            uint8_t  tx_search_skip_fag = (picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_ENC_DEC && (picture_control_set_ptr->parent_pcs_ptr->atb_mode == 0 || cu_ptr->prediction_mode_flag == INTER_MODE)) ? get_skip_tx_search_flag(
+            uint8_t  tx_search_skip_flag = (picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_ENC_DEC && (picture_control_set_ptr->parent_pcs_ptr->atb_mode == 0 || cu_ptr->prediction_mode_flag == INTER_MODE)) ? get_skip_tx_search_flag(
                 context_ptr->blk_geom->sq_size,
                 MAX_MODE_COST,
                 0,
                 1) : 1;
 
-            if (!tx_search_skip_fag) {
+            if (!tx_search_skip_flag) {
                     encode_pass_tx_search_hbd(
                         picture_control_set_ptr,
                         context_ptr,
@@ -1636,7 +1636,7 @@ void perform_intra_coding_loop(
         }
         // Encode Transform Unit -INTRA-
 
-        uint8_t cb_qp = cu_ptr->qp;
+        uint16_t cb_qp = cu_ptr->qp;
         Av1EncodeLoopFunctionTable[is16bit](
             picture_control_set_ptr,
             context_ptr,
@@ -1902,7 +1902,7 @@ void perform_intra_coding_loop(
         }
 
         // Encode Transform Unit -INTRA-
-        uint8_t cb_qp = cu_ptr->qp;
+        uint16_t cb_qp = cu_ptr->qp;
 
         Av1EncodeLoopFunctionTable[is16bit](
             picture_control_set_ptr,
@@ -2726,7 +2726,7 @@ EB_EXTERN void av1_encode_pass(
 
                             // Encode Transform Unit -INTRA-
                             {
-                                uint8_t             cb_qp = cu_ptr->qp;
+                                uint16_t             cb_qp = cu_ptr->qp;
 
                                 Av1EncodeLoopFunctionTable[is16bit](
                                     picture_control_set_ptr,
@@ -3091,7 +3091,7 @@ EB_EXTERN void av1_encode_pass(
 
                     uint32_t totTu = context_ptr->blk_geom->txb_count[cu_ptr->tx_depth];
                     uint8_t   tuIt;
-                    uint8_t   cb_qp = cu_ptr->qp;
+                    uint16_t   cb_qp = cu_ptr->qp;
                     uint32_t  component_mask = context_ptr->blk_geom->has_uv ? PICTURE_BUFFER_DESC_FULL_MASK : PICTURE_BUFFER_DESC_LUMA_MASK;
 
                     if (cu_ptr->prediction_unit_array[0].merge_flag == EB_FALSE) {

--- a/Source/Lib/Common/Codec/EbEncDecProcess.c
+++ b/Source/Lib/Common/Codec/EbEncDecProcess.c
@@ -257,7 +257,7 @@ static void ResetEncDec(
 #endif
     // Asuming cb and cr offset to be the same for chroma QP in both slice and pps for lambda computation
 
-    context_ptr->chroma_qp = context_ptr->qp;
+    context_ptr->chroma_qp = (uint8_t)context_ptr->qp;
 
     // Lambda Assignement
     context_ptr->qp_index = (uint8_t)picture_control_set_ptr->
@@ -296,7 +296,7 @@ static void EncDecConfigureLcu(
     context_ptr->qp = sb_qp;
 
     // Asuming cb and cr offset to be the same for chroma QP in both slice and pps for lambda computation
-    context_ptr->chroma_qp = context_ptr->qp;
+    context_ptr->chroma_qp = (uint8_t)context_ptr->qp;
     /* Note(CHKN) : when Qp modulation varies QP on a sub-LCU(CU) basis,  Lamda has to change based on Cu->QP , and then this code has to move inside the CU loop in MD */
     (void)sb_ptr;
     context_ptr->qp_index = (uint8_t)picture_control_set_ptr->

--- a/Source/Lib/Common/Codec/EbModeDecisionProcess.c
+++ b/Source/Lib/Common/Codec/EbModeDecisionProcess.c
@@ -396,7 +396,7 @@ void reset_mode_decision(
     context_ptr->qp = picture_control_set_ptr->picture_qp;
 #endif
     // Asuming cb and cr offset to be the same for chroma QP in both slice and pps for lambda computation
-    context_ptr->chroma_qp = context_ptr->qp;
+    context_ptr->chroma_qp = (uint8_t)context_ptr->qp;
     context_ptr->qp_index = (uint8_t)frm_hdr->quantization_params.base_q_idx;
     (*av1_lambda_assignment_function_table[picture_control_set_ptr->parent_pcs_ptr->pred_structure])(
         &context_ptr->fast_lambda,
@@ -454,7 +454,7 @@ void mode_decision_configure_lcu(
     context_ptr->qp = sb_qp;
     // Asuming cb and cr offset to be the same for chroma QP in both slice and pps for lambda computation
 
-    context_ptr->chroma_qp = context_ptr->qp;
+    context_ptr->chroma_qp = (uint8_t)context_ptr->qp;
 
     /* Note(CHKN) : when Qp modulation varies QP on a sub-LCU(CU) basis,  Lamda has to change based on Cu->QP , and then this code has to move inside the CU loop in MD */
 

--- a/Source/Lib/Common/Codec/EbPictureManagerProcess.c
+++ b/Source/Lib/Common/Codec/EbPictureManagerProcess.c
@@ -772,7 +772,7 @@ void* picture_manager_kernel(void *input_ptr)
                                     }
                                     // Set the Reference Object
                                     ChildPictureControlSetPtr->ref_pic_ptr_array[REF_LIST_0][refIdx] = referenceEntryPtr->reference_object_ptr;
-                                    ChildPictureControlSetPtr->ref_pic_qp_array[REF_LIST_0][refIdx] = ((EbReferenceObject*)referenceEntryPtr->reference_object_ptr->object_ptr)->qp;
+                                    ChildPictureControlSetPtr->ref_pic_qp_array[REF_LIST_0][refIdx] = (uint8_t)((EbReferenceObject*)referenceEntryPtr->reference_object_ptr->object_ptr)->qp;
                                     ChildPictureControlSetPtr->ref_slice_type_array[REF_LIST_0][refIdx] = ((EbReferenceObject*)referenceEntryPtr->reference_object_ptr->object_ptr)->slice_type;
                                     // Increment the Reference's liveCount by the number of tiles in the input picture
                                     eb_object_inc_live_count(

--- a/Source/Lib/Common/Codec/EbProductCodingLoop.c
+++ b/Source/Lib/Common/Codec/EbProductCodingLoop.c
@@ -2662,8 +2662,8 @@ void AV1CostCalcCfl(
     uint32_t                            chroma_height = context_ptr->blk_geom->bheight_uv;
     // FullLoop and TU search
     int32_t                             alpha_q3;
-    uint8_t                             cb_qp = context_ptr->qp;
-    uint8_t                             cr_qp = context_ptr->qp;
+    uint16_t                             cb_qp = context_ptr->qp;
+    uint16_t                             cr_qp = context_ptr->qp;
 
     full_distortion[DIST_CALC_RESIDUAL] = 0;
     full_distortion[DIST_CALC_PREDICTION] = 0;
@@ -3171,9 +3171,9 @@ uint8_t get_skip_tx_search_flag(
 {
     //NM: Skip tx search when the fast cost of the current mode candidate is substansially
     // Larger than the best fast_cost (
-    uint8_t  tx_search_skip_fag = cu_cost >= ((ref_fast_cost * weight) / 100) ? 1 : 0;
-    tx_search_skip_fag = sq_size >= 128 ? 1 : tx_search_skip_fag;
-    return tx_search_skip_fag;
+    uint8_t  tx_search_skip_flag = cu_cost >= ((ref_fast_cost * weight) / 100) ? 1 : 0;
+    tx_search_skip_flag = sq_size >= 128 ? 1 : tx_search_skip_flag;
+    return tx_search_skip_flag;
 }
 
 static INLINE PredictionMode get_uv_mode(UvPredictionMode mode) {
@@ -3627,17 +3627,17 @@ void perform_intra_tx_partitioning(
     uint64_t best_cost_search = (uint64_t)~0;
 
     TxType best_tx_type_depth_0 = DCT_DCT; // Track the best tx type @ depth 0 to be used @ the final stage (i.e. avoid redoing the tx type search).
-    uint8_t  tx_search_skip_fag;
+    uint8_t  tx_search_skip_flag;
     if (context_ptr->md_staging_tx_search == 0)
-        tx_search_skip_fag = EB_TRUE;
+        tx_search_skip_flag = EB_TRUE;
     else if (context_ptr->md_staging_tx_search == 1)
-        tx_search_skip_fag = picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_FULL_LOOP ? get_skip_tx_search_flag(
+        tx_search_skip_flag = picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_FULL_LOOP ? get_skip_tx_search_flag(
             context_ptr->blk_geom->sq_size,
             ref_fast_cost,
             *candidate_buffer->fast_cost_ptr,
             picture_control_set_ptr->parent_pcs_ptr->tx_weight) : EB_TRUE;
     else
-        tx_search_skip_fag = picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_FULL_LOOP ? EB_FALSE : EB_TRUE;
+        tx_search_skip_flag = picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_FULL_LOOP ? EB_FALSE : EB_TRUE;
 
     // Reset depth_1 neighbor arrays
     if (end_tx_depth) {
@@ -3743,7 +3743,7 @@ void perform_intra_tx_partitioning(
                 context_ptr->blk_geom->tx_height[context_ptr->tx_depth][context_ptr->txb_itr]);
 
             TxType best_tx_type = DCT_DCT;
-            if (!tx_search_skip_fag) {
+            if (!tx_search_skip_flag) {
             TxType txk_start = DCT_DCT;
             TxType txk_end = TX_TYPES;
             uint64_t best_cost_tx_search = (uint64_t)~0;
@@ -4405,19 +4405,19 @@ void full_loop_core(
                 context_ptr->blk_geom->bheight);
 
             // Transform partitioning free path
-            uint8_t  tx_search_skip_fag;
+            uint8_t  tx_search_skip_flag;
             if (context_ptr->md_staging_tx_search == 0)
-                tx_search_skip_fag = EB_TRUE;
+                tx_search_skip_flag = EB_TRUE;
             else if (context_ptr->md_staging_tx_search == 1)
-                tx_search_skip_fag = picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_FULL_LOOP ? get_skip_tx_search_flag(
+                tx_search_skip_flag = picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_FULL_LOOP ? get_skip_tx_search_flag(
                     context_ptr->blk_geom->sq_size,
                     ref_fast_cost,
                     *candidate_buffer->fast_cost_ptr,
                     picture_control_set_ptr->parent_pcs_ptr->tx_weight) : EB_TRUE;
             else
-                tx_search_skip_fag = picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_FULL_LOOP ? EB_FALSE : EB_TRUE;
+                tx_search_skip_flag = picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_FULL_LOOP ? EB_FALSE : EB_TRUE;
 
-            if (!tx_search_skip_fag) {
+            if (!tx_search_skip_flag) {
                 product_full_loop_tx_search(
                     candidate_buffer,
                     context_ptr,
@@ -4455,8 +4455,8 @@ void full_loop_core(
         cr_coeff_bits = 0;
 
         // FullLoop and TU search
-        uint8_t cb_qp = context_ptr->qp;
-        uint8_t cr_qp = context_ptr->qp;
+        uint16_t cb_qp = context_ptr->qp;
+        uint16_t cr_qp = context_ptr->qp;
         if (context_ptr->md_staging_skip_full_chroma == EB_FALSE) {
 
             if (context_ptr->blk_geom->has_uv && context_ptr->chroma_level <= CHROMA_MODE_1) {
@@ -4547,8 +4547,8 @@ void full_loop_core(
                         inputCbOriginIndex,
                         cuChromaOriginIndex,
                         candidate_buffer,
-                        cb_qp,
-                        cr_qp,
+                        (uint8_t)cb_qp,
+                        (uint8_t)cr_qp,
                         cbFullDistortion,
                         crFullDistortion,
                         &cb_coeff_bits,
@@ -4997,12 +4997,12 @@ void inter_depth_tx_search(
     EbAsm                                   asm_type)
 {
     // Hsan: if Transform Search ON and INTRA, then Tx Type search is performed @ the full loop
-    uint8_t  tx_search_skip_fag = (picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_INTER_DEPTH && (picture_control_set_ptr->parent_pcs_ptr->atb_mode == 0 || candidate_buffer ->candidate_ptr->type == INTER_MODE)) ? get_skip_tx_search_flag(
+    uint8_t  tx_search_skip_flag = (picture_control_set_ptr->parent_pcs_ptr->tx_search_level == TX_SEARCH_INTER_DEPTH && (picture_control_set_ptr->parent_pcs_ptr->atb_mode == 0 || candidate_buffer ->candidate_ptr->type == INTER_MODE)) ? get_skip_tx_search_flag(
         context_ptr->blk_geom->sq_size,
         ref_fast_cost,
         *candidate_buffer->fast_cost_ptr,
         picture_control_set_ptr->parent_pcs_ptr->tx_weight) : 1;
-    if (!tx_search_skip_fag) {
+    if (!tx_search_skip_flag) {
         uint64_t      y_full_distortion[DIST_CALC_TOTAL] = { 0 };
         uint32_t      count_non_zero_coeffs[3][MAX_NUM_OF_TU_PER_CU];
 
@@ -5055,8 +5055,8 @@ void inter_depth_tx_search(
         cr_coeff_bits = 0;
 
         // FullLoop and TU search
-        uint8_t cb_qp = context_ptr->qp;
-        uint8_t cr_qp = context_ptr->qp;
+        uint16_t cb_qp = context_ptr->qp;
+        uint16_t cr_qp = context_ptr->qp;
         if (context_ptr->blk_geom->has_uv && context_ptr->chroma_level <= CHROMA_MODE_1) {
             full_loop_r(
                 context_ptr->sb_ptr,
@@ -5598,8 +5598,8 @@ void search_best_independent_uv_mode(
                     context_ptr->blk_geom->txsize_uv[0][0],
                     frm_hdr->reduced_tx_set);
 
-            uint8_t  cb_qp = context_ptr->qp;
-            uint8_t  cr_qp = context_ptr->qp;
+            uint16_t  cb_qp = context_ptr->qp;
+            uint16_t  cr_qp = context_ptr->qp;
             uint64_t cb_coeff_bits = 0;
             uint64_t cr_coeff_bits = 0;
             uint64_t cbFullDistortion[DIST_CALC_TOTAL] = { 0, 0 };

